### PR TITLE
Add support to parse parquet files

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
This commit adds feature to accept parquet file as
1) target index dataset file 
2) neighbor dataset
3) query dataset . 
It also supports partition. For every partition, one iterator will be created.
Since, parquet file can be very big and it doesn't support index based loading, we will use batch iterator
and load one batch at a time and extract vectors for
given start and end offset. We will also use variable to hold batch if complete
batch is not processed. Once the batch is processed, it will clear the variable.

### Testing
- [x] New functionality includes testing

[Describe how this change was tested]

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/utils/dataset.py [hunk 1]
```diff
@@ -65,19 +66,22 @@ def reset(self):
...
 
-def get_data_set(data_set_format: str, path: str, context: Context):
+def get_data_set(data_set_format: str, path: str, context: Context = None, column_name: str = None):
     """
     Factory method to get instance of Dataset for given format.
     Args:
         data_set_format: File format like hdf5, bigann
         path: Data set file path
         context: Dataset Context Enum
+        column_name: Dataset column name that contains expected dataset
```
Comment (jmazanec15) [inline osbenchmark/utils/dataset.py:76]: contains vector field

File: osbenchmark/workload/params.py [hunk 2]
```diff
@@ -888,6 +888,7 @@ def __init__(self, workload, params, context: Context, **kwargs):
         self.data_set_format = parse_string_parameter("data_set_format", params)
         self.data_set_path = parse_string_parameter("data_set_path", params, "")
         self.data_set_corpus = parse_string_parameter("data_set_corpus", params, "")
+        self.data_set_name = params.get("data_set_name")
```
Comment (jmazanec15) [inline osbenchmark/workload/params.py:891]: should we use parse_string_parameter call as done above? Also, how does this relate to this change?

File: osbenchmark/utils/dataset.py [hunk 3]
```diff
@@ -144,6 +148,115 @@ def parse_context(context: Context) -> str:
...
+        @param end_offset: start offset to extract vectors from parquet file
+        @return: array of vectors
+        """
+        vectors = []
+        while True:
+            batch = self.get_batch()
+            if not batch:
+                break
+            batch_length = len(batch)
+            if start_offset > self.file_batch_end:
```
Comment (jmazanec15) [inline osbenchmark/utils/dataset.py:196]: Why do we need this? Cant we seek?

**Review by jmazanec15 — COMMENTED** [review]
Overall, minor questions, but looks good. Are we documenting usage anywhere?

File: osbenchmark/workload/params.py (hunk 2 above)
Comment (VijayanB) [reply osbenchmark/workload/params.py:891]: To read parquet, we need the column name from users. Based on my observation, i see different column is used to store vectors in parquet file.

File: osbenchmark/utils/dataset.py (hunk 3 above)
Comment (VijayanB) [reply osbenchmark/utils/dataset.py:196]: I couldn't find a way to seek parquet file. Hence, i have to use batch to perform in-memory seek.

**Comment by IanHoang** [comment]
Closing this PR. Feel free to reopen when ready.

**Comment by VijayanB** [comment]
Reopening this PR. Will work on this PR for this release

**Comment by IanHoang** [comment]
@VijayanB Please convert to draft or reopen if still working on it. We're aiming to have the next release a couple weeks after [the next OpenSearch minor release](https://opensearch.org/releases.html) (2.16.0).

**Pull request closed without merging**